lib/msal-browser/src/naa/BridgeProxy.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AccountInfo } from "./AccountInfo";
+import {
+    AccountByHomeIdRequest,
+    AccountByLocalIdRequest,
+    AccountByUsernameRequest,
+} from "./AccountRequests";
+import { AuthBridge } from "./AuthBridge";
+import { BridgeCapabilities } from "./BridgeCapabilities";
+import { BridgeRequest } from "./BridgeRequest";
+import { BridgeRequestEnvelope, BridgeMethods } from "./BridgeRequestEnvelope";
+import { BridgeResponseEnvelope } from "./BridgeResponseEnvelope";
+import { IBridgeProxy } from "./IBridgeProxy";
+import { InitializeBridgeResponse } from "./InitializeBridgeResponse";
+import { TokenRequest } from "./TokenRequest";
+import { TokenResponse } from "./TokenResponse";
+
+declare global {
+    interface Window {
+        nestedAppAuthBridge: AuthBridge;
+    }
+}
+
+/**
+ * BridgeProxy
+ * Provides a proxy for accessing a bridge to a host app and/or
+ * platform broker
+ */
+export class BridgeProxy implements IBridgeProxy {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    static bridgeRequests: any[] = [];
+    static crypto: Crypto;
+    sdkName: string;
+    sdkVersion: string;
+    capabilities: BridgeCapabilities;
+
+    /**
+     * initializeNestedAppAuthBridge - Initializes the bridge to the host app
+     * @returns a promise that resolves to an InitializeBridgeResponse or rejects with an Error
+     * @remarks This method will be called by the create factory method
+     * @remarks If the bridge is not available, this method will throw an error
+     */
+    protected static async initializeNestedAppAuthBridge(): Promise<InitializeBridgeResponse> {
+        if (window === undefined) {
+            throw new Error("window is undefined");
+        }
+        if (window.nestedAppAuthBridge === undefined) {
+            throw new Error("window.nestedAppAuthBridge is undefined");
+        }
+        if (window.crypto === undefined) {
+            throw new Error("window.crypto is undefined");
+        }
+
+        try {
+            BridgeProxy.crypto = window.crypto;
+
+            window.nestedAppAuthBridge.addEventListener(
+                "message",
+                (response: string) => {
+                    window.console.log(response);
+                    const responseEnvelope: BridgeResponseEnvelope =
```

Comment:
This is a force cast. Do we instead need to massage/map the response to _create_ a new responseEnvelope so that we can ensure that consumers get the response body they'd expect?